File: BOJ/11723번/11723.cpp
#include <iostream>
#include <string>
using namespace std;

int main(void)
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int M, num;
    cin >> M;

    string opers;
    int target = 0;

    while (M--)
    {
        cin >> opers;

        if (opers.compare("add") == 0)
        {
            cin >> num;
            target |= (1 << (num - 1));
        }
        else if (opers.compare("remove") == 0)
        {
            cin >> num;
            target &= ~(1 << (num - 1));
        }
        else if (opers.compare("check") == 0)
        {
            cin >> num;
            int check = target & (1 << (num - 1));
            cout << (check != 0) << '\n';
        }
        else if (opers.compare("toggle") == 0)
        {
            cin >> num;
            target ^= (1 << (num - 1));
        }
        else if (opers.compare("all") == 0)
        {
            target = ~(1 << 20);
        }
        else // empty
        {
            target = 0;
        }
    }

    return 0;
}

File: BOJ/11723번/11723.kt
import java.util.*
fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()
    var number = 0
    val m = br.readLine().toInt()
    val result = StringBuilder()
    for (it in 0 until m) {
        val token = StringTokenizer(br.readLine())
        when (token.nextToken()) {
            "add" -> {
                val num = (1 shl (token.nextToken().toInt() - 1))
                if (number and num == 0) number = number or num
            }
            "remove" -> {
                val num = (1 shl (token.nextToken().toInt() - 1))
                if (number and num == num) number = number and num.inv()
            }
            "check" -> {
                val num = (1 shl (token.nextToken().toInt() - 1))
                result.append(
                    when (number and num == num) {
                        true -> "1\n"
                        false -> "0\n"
                    }
                )
            }
            "toggle" -> {
                val num = (1 shl (token.nextToken().toInt() - 1))
                number =
                    if (number and num == num) number and num.inv()
                    else number or num
            }
            "all" -> number = (1 shl 21) - 1
            "empty" -> number = 0
        }
    }
    bw.write(result.toString())
    bw.close()
}